### PR TITLE
Cleanup Protocol classes

### PR DESCRIPTION
**Pull request by tigerw**

### CODE REVIEW

File: src/Protocol/Protocol_1_8.h [hunk 1]
```diff
@@ -246,6 +229,9 @@ class cProtocol_1_8_0:
 	/** Writes the block entity data for the specified block entity into the packet. */
 	virtual void WriteBlockEntity(cPacketizer & a_Pkt, const cBlockEntity & a_BlockEntity);
 
+	/** State of the protocol. 1 = status, 2 = login, 3 = game */
+	UInt32 m_State;
```
Comment (peterbell10) [inline src/Protocol/Protocol_1_8.h:233]: Variable declarations go above functions. This is the way.

Comment (tigerw) [reply src/Protocol/Protocol_1_8.h:233]: I now know de wey, thank.

File: src/Protocol/Protocol_1_14.cpp [hunk 2]
```diff
@@ -215,77 +215,77 @@ cProtocol::Version cProtocol_1_14::GetProtocolVersion()
...
 
 
 
 
 
-std::pair<short, short> cProtocol_1_14::GetItemFromProtocolID(UInt32 a_ProtocolID)
+bool cProtocol_1_14::HandlePacket(cByteBuffer & a_ByteBuffer, UInt32 a_PacketType)
 {
-	return PaletteUpgrade::ToItem(Palette_1_14::ToItem(a_ProtocolID));
+	if (m_State != 3)
```
Comment (NiLSPACE) [inline src/Protocol/Protocol_1_14.cpp:256]: Magic number

Comment (peterbell10) [reply src/Protocol/Protocol_1_14.cpp:256]: He's just alpha-sorting the functions I think. That magic number was always there.

Comment (tigerw) [reply src/Protocol/Protocol_1_14.cpp:256]: muahaha! if you think i'm only alpha sorting then my EVIL PLAN has succeeded!!1!

anyway I can also make it an enum, it's a cleanup anyway.

Comment (tigerw) [reply src/Protocol/Protocol_1_14.cpp]: made it a plain enum. Scoped enum will be a mass-replace later due to all the assert(m_State == 3) everywhere.

**Comment by tigerw** [comment]
CI broke, retriggered

**Comment by tigerw** [comment]
passes

**Comment by tigerw** [comment]
merge?

File: src/Protocol/Protocol_1_8.cpp [hunk 3]
```diff
@@ -100,10 +100,10 @@ extern bool g_ShouldLogCommIn, g_ShouldLogCommOut;
 ////////////////////////////////////////////////////////////////////////////////
 // cProtocol_1_8_0:
 
-cProtocol_1_8_0::cProtocol_1_8_0(cClientHandle * a_Client, const AString & a_ServerAddress, UInt16 a_ServerPort, UInt32 a_State) :
+cProtocol_1_8_0::cProtocol_1_8_0(cClientHandle * a_Client, const AString & a_ServerAddress, UInt16 a_ServerPort, State a_State) :
 	Super(a_Client),
 	m_ServerAddress(a_ServerAddress),
-	m_ServerPort(a_ServerPort),
+	// Note: a_ServerPort is unused
```
Comment (bearbin) [inline src/Protocol/Protocol_1_8.cpp:106]: If it's unused, why not remove it from the constructor?

Comment (tigerw) [reply src/Protocol/Protocol_1_8.cpp:106]: Come to think of it I made all the constructors 1.9+ inherit so this change shouldn't need to touch every file.

**Comment by tigerw** [comment]
Ready. Intention to rebase, not squish

**Review by bearbin — APPROVED** [review]

**Comment by bearbin** [comment]
Since the intention is to rebase rather than squish, this probably needs a force push after rebasing on top of master...

**Pull request merged**